bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -210,6 +210,32 @@ export class PlotCanvasView extends DOMView
     follow_enabled = false
     has_bounds = false
 
+    if @model.plot.aspect_ratio != false
+      aspect = if @model.plot.aspect_ratio is true then 1 else @model.plot.aspect_ratio
+      r = aspect*(@model.plot.width/@model.plot.height)
```

Comment:
Pretty sure this should be using the plot frame's "inner width" and "inner height", those are the pixel dimensions of the area between the axes, i.e.  not counting any space taken up by title, axes, tick labels, padding, etc.